scripts/building/check_enforcer.ps1

```diff
@@ -0,0 +1,181 @@
+                Write-Output "check runs name $($_.name)"
+            }
+            foreach ($key in $pipelines.Keys) {
+                $value = $pipelines[$key]
+                if ($value -eq 0) {
+                    continue
+                }
+                if ($_.name.Contains($key)) {
+                    $pipelines_count[$key] += 1
+                    $valid_status_array_ref.Value += $_
+                }
+            }
+        }
+    }
+    # Get pipelines of commit. count should match.
+    foreach ($key in $pipelines.Keys) {
+        if ($pipelines_count[$key] -lt $pipelines[$key]) {
+            $failed_reason_ref.Value = "Not all pipelines are triggered."
+        }
+    }
+}
+function sdk_cli_checks([ref]$failed_reason_ref, $valid_status_array) {
+    $failed_reason_ref.Value = ""
+    $pipelines = @{
+        "sdk_cli_tests" = 4;
+        "sdk_cli_global_config_tests" = 1;
+        "sdk_cli_azure_test" = 4;
+    }
+
+    $pipelines_success_count = @{
+        "sdk_cli_tests" = 0;
+        "sdk_cli_global_config_tests" = 0;
+        "sdk_cli_azure_test" = 0;
+    }
+    
+    $valid_status_array `
+    | ForEach-Object {
+        foreach ($key in $pipelines.Keys) {
+            $value = $pipelines[$key]
+            if ($value -eq 0) {
+                continue
+            }
+            if ($_.name.Contains($key)) {
+                if ($_.conclusion -ieq "success") {
+                    $pipelines_success_count[$key] += 1
+                } elseif ($_.conclusion -ieq "failure") {
+                    $failed_reason_ref.Value = "Required pipelines are not successful."
+                } else {
+                    if ($failed_reason_ref.Value -eq "") {
+                        $failed_reason_ref.Value = "Required pipelines are not finished."
+                    }
+                }
+            }
+            Write-Output "$($_.name) is $($_.conclusion)."
+        }
+    }
+}
+
+function run_checks() {
+
+    $need_to_check = New-Object System.Collections.Generic.HashSet[string]
+    git diff --name-only HEAD origin/main | ForEach-Object {
+        if ($snippet_debug -eq 1) {
+            Write-Output "git diff --name-only HEAD main $_"
+        }
+        if ($_.Contains("src/promptflow")) {
+            $need_to_check.Add("sdk_cli")
+        }
+    }
+
+    $sdk_cli_check = $false
+
+    foreach ($item in $need_to_check) {
+        if ($item -eq "sdk_cli") {
+            $sdk_cli_check = $true
+        }
+    }
+
+    $failed_reason =  ""
+    
+    for ($i = 0; $i -lt $LoopTimes; $i++) {
+        Start-Sleep -Seconds 30
+
+        $failed_reason = ""
+        $valid_status_array = @()
+
+        # Get all triggered pipelines.
+        # If not all pipelines are triggered, continue.
+        if ($sdk_cli_check -eq $true) {
+            sdk_cli_trigger_checks ([ref]$failed_reason) ([ref]$valid_status_array)
+        }
+        if ($failed_reason -ne "") {
+            Write-Output "$failed_reason"
```

Comment:
I want to understand more on this line: if `failed_reason` here is not empty, I think SDK/CLI CIs are not triggered correctly? then we still need to wait for some minutes? - am I understanding correctly on this? if so, shall we make it a quick fail?

Reply:
oh, I think maybe it's kind of difficult to identity not trigger yet AND not trigger?

Reply:
Yes, let me make this situation quicker.

Reply:
All pipelines, should trigger within 3 minutes (in our cases), I'll make a counter here.